### ISSUE TITLE
Polyhedron demo:  Ctrl - A 

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#4772]
## Issue Details

When I create a grid of 20x20x20 spheres with _Create Grid..._    and then type Ctrl-A  to select them all this takes forever.

## Environment

* Operating system: Windows 
* Compiler: VC++
* Release or debug mode: Release
* Specific flags used (if any):
* CGAL version: master
* Boost version:
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by afabri** [comment]
This has nothing to do with Create Grid.  This only generates quickly many objects.

**Comment by afabri** [comment]
And it is the same for _Erase All_

**Comment by maxGimeno** [comment]
To be able to use and draw groups, we use a QTreeView to display the scene (the item names), with a proxy to be able to search  in the tree for a specific item. 
After manipulating a simple example from Qt to use such a QTreeView, I can affirm that the freeze that happens when the user ask for a global selection is due to this QTreeView. The example I use is only storing names and address in a virtual address book. If I creat 8000 entries called A0-A7999 with no address, and then select them all, it freezes for around 15 sec. 
The original view for this example is a QTableView, but the use of groups require specifically the QTreeView, to be able to draw items that are not aligned, with a hierarchy.

So my conclusion is that I can't "fix" that. A possibility would be to optimize when there are no groups, but I'm not sure it would be worth the effort.

**Comment by lrineau** [comment]
You could test `proxyModel->filterRegExp()->isEmpty()`, to avoid passing in `proxyModel->mapSelectionFromSource(..)` if the regexpr is empty. Maybe that could speed-up the selection when the filter is empty. The implementation of `mapSelectionFromSource` in Qt5 seems to be rather naive and inefficient.

**Comment by afabri** [comment]
I suggested to look at [`QTreeView::selectAll()`](https://doc.qt.io/qt-5/qtreeview.html#selectAll)

**Comment by lrineau** [comment]
I had a look at the code of `QTreeView::selectAll()`. If there is not filtering proxy, that looks like what we do.

The code could be something like:
```c++
  if(proxyModel->filterRegExp()->isEmpty()) {
    sceneView->selectAll();
  } else {
    // the code as it is now
    QItemSelection s =
      proxyModel->mapSelectionFromSource(scene->createSelectionAll());
    sceneView->selectionModel()->select(s,
                                        QItemSelectionModel::ClearAndSelect);
  }
```

**Comment by maxGimeno** [comment]
oh because sceneView->selectAll() doesn't care for the search, is that it ? That would explain why we didn't use it.

**Comment by maxGimeno** [comment]
No, it works the same with sceneView->selectAll(); No need to dichotomize.

**Comment by lrineau** [comment]
Closed by the merge of #4949.
